data_structures/linked_list/singly_linked_list.py

```diff
@@ -10,46 +10,54 @@ def __repr__(self):
 class LinkedList:
     def __init__(self):
         self.head = None  # initialize head to None
+        self.size = 0  # length of linked list
```

Comment:
I am not a fan of this change.  This creates duplicate state and opens us up to too many locations where this variable must be accurately updated or we have bugs that are difficult to debug.

Instead, please consider creating an [`.__iter__()`](https://docs.python.org/3/reference/datamodel.html#object.__iter__) method.  Then `__len__()` becomes `return len(tuple(self))`.  This could also simplify  other methods like `__repr__()`, `__str__()`, etc.  There are examples of creating this method elsewhere in this repo.

Reply:
Can you please give me example file link in this repo.

Reply:
https://github.com/TheAlgorithms/Python/search?q=__iter__

Reply:
```python
def __iter__(self):
        node = self.head
        while node:
            yield node.data
            node = node.next
```

Reply:
add this add source file ?

Reply:
Maybe it is just me but I like to sort the `.__xxx__()` methods above the other methods so the reader knows the builtin capabilities before reading the custom methods.

Reply:
Could you please clone this this branch and commit what you what. I am not similar to implement `__iter__` . Thanks in advance.